src/System.Memory/src/System/Buffers/Text/Utf8Formatter.Boolean.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers.Text
+{
+    public static partial class Utf8Formatter
+    {
+        /// <summary>
+        /// Formats a Boolean as a UTF8 string.
+        /// </summary>
+        /// <param name="value">Value to format</param>
+        /// <param name="buffer">Buffer to receive UTF8 string</param>
+        /// <param name="bytesWritten">Receives the length of the formatted text in bytes</param>
+        /// <param name="format">The standard format to use</param>
+        /// <returns>
+        /// true for success. "bytesWritten" contains the length of the formatted test in bytes.
+        /// false if buffer was too short. Iteratively increase the size of the buffer and retry until it succeeds. 
+        /// </returns>
+        /// <exceptions>
+        /// <cref>System.FormatException</cref> if the format is not valid for this data type.
+        /// </exceptions>
+        public static bool TryFormat(bool value, Span<byte> buffer, out int bytesWritten, StandardFormat format = default)
+        {
+            bool upperCase;
+            if (format.IsDefault || format.Symbol == 'G')
+            {
+                upperCase = true;
+            }
+            else if (format.Symbol == 'l')
+            {
+                upperCase = false;
+            }
+            else
+            {
+                throw new FormatException(SR.Argument_BadFormatSpecifier);
+            }
+
+            if (value)
+            {
+                if (buffer.Length < 4) // 4 bytes for "true"
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+
+                byte[] result = upperCase ? ParserFormatterConstants.s_True : ParserFormatterConstants.s_true;
+                result.CopyTo(buffer);
```

Comment:
You might want to do TryCopyTo. Right now, you do the range check and then CopyTo does another one.